diff --git a/AK/ByteBuffer.h b/AK/ByteBuffer.h
index f1d57e3e33..451c23469c 100644
--- a/AK/ByteBuffer.h
+++ b/AK/ByteBuffer.h
@@ -285,8 +285,11 @@ public:
     void overwrite(size_t offset, void const* data, size_t data_size)
     {
         // make sure we're not told to write past the end
+#pragma GCC diagnostic push
+#pragma GCC diagnostic warning "-Wstringop-overflow"
         VERIFY(offset + data_size <= size());
         __builtin_memmove(this->data() + offset, data, data_size);
+#pragma GCC diagnostic pop
     }
 
     void zero_fill()
