diff --git a/Userland/Libraries/LibWasm/CMakeLists.txt b/Userland/Libraries/LibWasm/CMakeLists.txt
index 729156278c..163b7dd9cb 100644
--- a/Userland/Libraries/LibWasm/CMakeLists.txt
+++ b/Userland/Libraries/LibWasm/CMakeLists.txt
@@ -11,6 +11,4 @@ set(SOURCES
 serenity_lib(LibWasm wasm)
 target_link_libraries(LibWasm PRIVATE LibCore LibJS)
 
-# FIXME: Install these into usr/Tests/LibWasm
 include(wasm_spec_tests)
-install(DIRECTORY Tests/ DESTINATION home/anon/Tests/wasm-tests)
