diff --git a/Meta/Lagom/Tools/CMakeLists.txt b/Meta/Lagom/Tools/CMakeLists.txt
index 68a854062b..21fb53842e 100644
--- a/Meta/Lagom/Tools/CMakeLists.txt
+++ b/Meta/Lagom/Tools/CMakeLists.txt
@@ -1,14 +1,16 @@
 function(lagom_tool tool)
-    cmake_parse_arguments(LAGOM_TOOL "" "" "SOURCES;LIBS" ${ARGN})
+    cmake_parse_arguments(LAGOM_TOOL "" "INSTALL" "SOURCES;LIBS" ${ARGN})
     add_executable(${tool} ${SOURCES} ${LAGOM_TOOL_SOURCES})
     # alias for parity with exports
     add_executable(Lagom::${tool} ALIAS ${tool})
     target_link_libraries(${tool} AK LibCoreMinimal LibFileSystem GenericClangPlugin ${LAGOM_TOOL_LIBS})
-    install(
-        TARGETS ${tool}
-        EXPORT LagomTargets
-        RUNTIME COMPONENT Lagom_Runtime
-    )
+    if (DEFINED LAGOM_TOOL_INSTALL AND LAGOM_TOOL_INSTALL)
+        install(
+            TARGETS ${tool}
+            EXPORT LagomTargets
+            RUNTIME COMPONENT Lagom_Runtime
+        )
+    endif()
 endfunction()
 
 add_subdirectory(CodeGenerators)
diff --git a/Meta/Lagom/Tools/CodeGenerators/IPCCompiler/CMakeLists.txt b/Meta/Lagom/Tools/CodeGenerators/IPCCompiler/CMakeLists.txt
index 9d41894503..d4f5afbbb6 100644
--- a/Meta/Lagom/Tools/CodeGenerators/IPCCompiler/CMakeLists.txt
+++ b/Meta/Lagom/Tools/CodeGenerators/IPCCompiler/CMakeLists.txt
@@ -2,4 +2,4 @@ set(SOURCES
     main.cpp
 )
 
-lagom_tool(IPCCompiler LIBS LibMain)
+lagom_tool(IPCCompiler LIBS LibMain INSTALL NO)
diff --git a/Meta/Lagom/Tools/CodeGenerators/LibTextCodec/CMakeLists.txt b/Meta/Lagom/Tools/CodeGenerators/LibTextCodec/CMakeLists.txt
index 32d9aced25..06e25a40a5 100644
--- a/Meta/Lagom/Tools/CodeGenerators/LibTextCodec/CMakeLists.txt
+++ b/Meta/Lagom/Tools/CodeGenerators/LibTextCodec/CMakeLists.txt
@@ -1 +1 @@
-lagom_tool(GenerateEncodingIndexes SOURCES GenerateEncodingIndexes.cpp LIBS LibMain)
+lagom_tool(GenerateEncodingIndexes SOURCES GenerateEncodingIndexes.cpp LIBS LibMain INSTALL NO)
diff --git a/Meta/Lagom/Tools/CodeGenerators/LibUnicode/CMakeLists.txt b/Meta/Lagom/Tools/CodeGenerators/LibUnicode/CMakeLists.txt
index a03d4cda73..5253994bb7 100644
--- a/Meta/Lagom/Tools/CodeGenerators/LibUnicode/CMakeLists.txt
+++ b/Meta/Lagom/Tools/CodeGenerators/LibUnicode/CMakeLists.txt
@@ -1,1 +1,1 @@
-lagom_tool(GenerateEmojiData SOURCES GenerateEmojiData.cpp LIBS LibMain)
+lagom_tool(GenerateEmojiData SOURCES GenerateEmojiData.cpp LIBS LibMain INSTALL NO)
diff --git a/Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/CMakeLists.txt b/Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/CMakeLists.txt
index 98e74e4583..7c84c74a9b 100644
--- a/Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/CMakeLists.txt
+++ b/Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/CMakeLists.txt
@@ -3,5 +3,5 @@ set(SOURCES
     main.cpp
 )
 
-lagom_tool(BindingsGenerator LIBS LibIDL LibMain)
+lagom_tool(BindingsGenerator LIBS LibIDL LibMain INSTALL NO)
 target_compile_options(BindingsGenerator PUBLIC -g)
diff --git a/Meta/Lagom/Tools/CodeGenerators/LibWeb/CMakeLists.txt b/Meta/Lagom/Tools/CodeGenerators/LibWeb/CMakeLists.txt
index 908aab6854..6fdd78df06 100644
--- a/Meta/Lagom/Tools/CodeGenerators/LibWeb/CMakeLists.txt
+++ b/Meta/Lagom/Tools/CodeGenerators/LibWeb/CMakeLists.txt
@@ -1,13 +1,13 @@
 set(SOURCES "") # avoid pulling SOURCES from parent scope
 
-lagom_tool(GenerateCSSEnums                 SOURCES GenerateCSSEnums.cpp LIBS LibMain)
-lagom_tool(GenerateCSSMathFunctions         SOURCES GenerateCSSMathFunctions.cpp LIBS LibMain)
-lagom_tool(GenerateCSSMediaFeatureID        SOURCES GenerateCSSMediaFeatureID.cpp LIBS LibMain)
-lagom_tool(GenerateCSSPropertyID            SOURCES GenerateCSSPropertyID.cpp LIBS LibMain)
-lagom_tool(GenerateCSSPseudoClass           SOURCES GenerateCSSPseudoClass.cpp LIBS LibMain)
-lagom_tool(GenerateCSSTransformFunctions    SOURCES GenerateCSSTransformFunctions.cpp LIBS LibMain)
-lagom_tool(GenerateCSSValueID               SOURCES GenerateCSSValueID.cpp LIBS LibMain)
-lagom_tool(GenerateWindowOrWorkerInterfaces SOURCES GenerateWindowOrWorkerInterfaces.cpp LIBS LibMain LibIDL)
-lagom_tool(GenerateAriaRoles                SOURCES GenerateAriaRoles.cpp LIBS LibMain)
+lagom_tool(GenerateCSSEnums                 SOURCES GenerateCSSEnums.cpp LIBS LibMain INSTALL NO)
+lagom_tool(GenerateCSSMathFunctions         SOURCES GenerateCSSMathFunctions.cpp LIBS LibMain INSTALL NO)
+lagom_tool(GenerateCSSMediaFeatureID        SOURCES GenerateCSSMediaFeatureID.cpp LIBS LibMain INSTALL NO)
+lagom_tool(GenerateCSSPropertyID            SOURCES GenerateCSSPropertyID.cpp LIBS LibMain INSTALL NO)
+lagom_tool(GenerateCSSPseudoClass           SOURCES GenerateCSSPseudoClass.cpp LIBS LibMain INSTALL NO)
+lagom_tool(GenerateCSSTransformFunctions    SOURCES GenerateCSSTransformFunctions.cpp LIBS LibMain INSTALL NO)
+lagom_tool(GenerateCSSValueID               SOURCES GenerateCSSValueID.cpp LIBS LibMain INSTALL NO)
+lagom_tool(GenerateWindowOrWorkerInterfaces SOURCES GenerateWindowOrWorkerInterfaces.cpp LIBS LibMain LibIDL INSTALL NO)
+lagom_tool(GenerateAriaRoles                SOURCES GenerateAriaRoles.cpp LIBS LibMain INSTALL NO)
 
 add_subdirectory(BindingsGenerator)
diff --git a/Meta/Lagom/Tools/CodeGenerators/LibWebView/CMakeLists.txt b/Meta/Lagom/Tools/CodeGenerators/LibWebView/CMakeLists.txt
index 35dec51cc6..e7dcaa9872 100644
--- a/Meta/Lagom/Tools/CodeGenerators/LibWebView/CMakeLists.txt
+++ b/Meta/Lagom/Tools/CodeGenerators/LibWebView/CMakeLists.txt
@@ -1 +1 @@
-lagom_tool(GeneratePublicSuffixData SOURCES GeneratePublicSuffixData.cpp LIBS LibMain)
+lagom_tool(GeneratePublicSuffixData SOURCES GeneratePublicSuffixData.cpp LIBS LibMain INSTALL NO)
