diff --git a/Tests/LibAudio/CMakeLists.txt b/Tests/LibAudio/CMakeLists.txt
index ae8d4a2910..98574e2abc 100644
--- a/Tests/LibAudio/CMakeLists.txt
+++ b/Tests/LibAudio/CMakeLists.txt
@@ -7,7 +7,3 @@ set(TEST_SOURCES
 foreach(source IN LISTS TEST_SOURCES)
     serenity_test("${source}" LibAudio LIBS LibAudio LibFileSystem)
 endforeach()
-
-install(DIRECTORY ${FLAC_SPEC_TEST_PATH} DESTINATION usr/Tests/LibAudio/FLAC)
-
-install(DIRECTORY WAV DESTINATION usr/Tests/LibAudio)
diff --git a/Tests/LibCompress/CMakeLists.txt b/Tests/LibCompress/CMakeLists.txt
index 5ce4dacdc9..91e9980759 100644
--- a/Tests/LibCompress/CMakeLists.txt
+++ b/Tests/LibCompress/CMakeLists.txt
@@ -12,6 +12,3 @@ set(TEST_SOURCES
 foreach(source IN LISTS TEST_SOURCES)
     serenity_test("${source}" LibCompress LIBS LibCompress)
 endforeach()
-
-install(DIRECTORY brotli-test-files DESTINATION usr/Tests/LibCompress)
-install(DIRECTORY deflate-test-files DESTINATION usr/Tests/LibCompress)
diff --git a/Tests/LibCore/CMakeLists.txt b/Tests/LibCore/CMakeLists.txt
index a159bf2c21..132036e644 100644
--- a/Tests/LibCore/CMakeLists.txt
+++ b/Tests/LibCore/CMakeLists.txt
@@ -19,5 +19,3 @@ target_link_libraries(TestLibCorePromise PRIVATE LibThreading)
 # NOTE: Required because of the LocalServer tests
 target_link_libraries(TestLibCoreStream PRIVATE LibThreading)
 target_link_libraries(TestLibCoreSharedSingleProducerCircularQueue PRIVATE LibThreading)
-
-install(FILES long_lines.txt 10kb.txt small.txt DESTINATION usr/Tests/LibCore)
diff --git a/Tests/LibGfx/CMakeLists.txt b/Tests/LibGfx/CMakeLists.txt
index 225246101f..05c38ceb45 100644
--- a/Tests/LibGfx/CMakeLists.txt
+++ b/Tests/LibGfx/CMakeLists.txt
@@ -18,5 +18,3 @@ set(TEST_SOURCES
 foreach(source IN LISTS TEST_SOURCES)
     serenity_test("${source}" LibGfx LIBS LibGfx)
 endforeach()
-
-install(DIRECTORY test-inputs DESTINATION usr/Tests/LibGfx)
diff --git a/Tests/LibJS/CMakeLists.txt b/Tests/LibJS/CMakeLists.txt
index 71075fc3dc..a91be71149 100644
--- a/Tests/LibJS/CMakeLists.txt
+++ b/Tests/LibJS/CMakeLists.txt
@@ -1,7 +1,5 @@
 serenity_testjs_test(test-js.cpp test-js LIBS LibLocale)
 
-install(TARGETS test-js RUNTIME DESTINATION bin OPTIONAL)
-
 serenity_test(test-invalid-unicode-js.cpp LibJS LIBS LibJS LibLocale)
 
 serenity_test(test-value-js.cpp LibJS LIBS LibJS LibLocale)
@@ -9,9 +7,7 @@ serenity_test(test-value-js.cpp LibJS LIBS LibJS LibLocale)
 add_executable(test262-runner test262-runner.cpp)
 target_link_libraries(test262-runner PRIVATE LibJS LibCore LibLocale)
 serenity_set_implicit_links(test262-runner)
-install(TARGETS test262-runner RUNTIME DESTINATION bin OPTIONAL)
 
 add_executable(test-test262 test-test262.cpp)
 target_link_libraries(test-test262 PRIVATE LibMain LibCore LibFileSystem)
 serenity_set_implicit_links(test-test262)
-install(TARGETS test-test262 RUNTIME DESTINATION bin OPTIONAL)
diff --git a/Tests/LibVideo/CMakeLists.txt b/Tests/LibVideo/CMakeLists.txt
index c2c4373b9e..40e9d38614 100644
--- a/Tests/LibVideo/CMakeLists.txt
+++ b/Tests/LibVideo/CMakeLists.txt
@@ -6,13 +6,3 @@ set(TEST_SOURCES
 foreach(source IN LISTS TEST_SOURCES)
     serenity_test("${source}" LibVideo LIBS LibVideo)
 endforeach()
-
-install(FILES vp9_in_webm.webm DESTINATION usr/Tests/LibVideo)
-install(FILES vp9_4k.webm DESTINATION usr/Tests/LibVideo)
-install(FILES vp9_clamp_reference_mvs.webm DESTINATION usr/Tests/LibVideo)
-install(FILES vp9_oob_blocks.webm DESTINATION usr/Tests/LibVideo)
-install(FILES master_elements_containing_crc32.mkv DESTINATION usr/Tests/LibVideo)
-install(FILES oss-fuzz-testcase-52630.vp9 DESTINATION usr/Tests/LibVideo)
-install(FILES oss-fuzz-testcase-53977.vp9 DESTINATION usr/Tests/LibVideo)
-install(FILES oss-fuzz-testcase-62054.vp9 DESTINATION usr/Tests/LibVideo)
-install(FILES oss-fuzz-testcase-63182.vp9 DESTINATION usr/Tests/LibVideo)
diff --git a/Tests/LibWasm/CMakeLists.txt b/Tests/LibWasm/CMakeLists.txt
index b6cf817796..85f6154026 100644
--- a/Tests/LibWasm/CMakeLists.txt
+++ b/Tests/LibWasm/CMakeLists.txt
@@ -1,2 +1 @@
 serenity_testjs_test(test-wasm.cpp test-wasm LIBS LibWasm LibJS LibCrypto)
-install(TARGETS test-wasm RUNTIME DESTINATION bin OPTIONAL)
diff --git a/Tests/LibWeb/CMakeLists.txt b/Tests/LibWeb/CMakeLists.txt
index 65268e0a40..dad1c094b8 100644
--- a/Tests/LibWeb/CMakeLists.txt
+++ b/Tests/LibWeb/CMakeLists.txt
@@ -14,5 +14,3 @@ foreach(source IN LISTS TEST_SOURCES)
 endforeach()
 
 target_link_libraries(TestFetchURL PRIVATE LibURL)
-
-install(FILES tokenizer-test.html DESTINATION usr/Tests/LibWeb)
