diff --git a/Userland/Libraries/LibJS/Runtime/Value.h b/Userland/Libraries/LibJS/Runtime/Value.h
index f1ebf841d3..2b6eb76297 100644
--- a/Userland/Libraries/LibJS/Runtime/Value.h
+++ b/Userland/Libraries/LibJS/Runtime/Value.h
@@ -439,7 +439,7 @@ public:
         // For AArch64 the top 16 bits of the pointer should be zero.
         return static_cast<FlatPtr>(encoded & 0xffff'ffff'ffffULL);
 #else
-#    error "Unknown architecture. Don't know whether pointers need to be sign-extended."
+        return static_cast<FlatPtr>(encoded);
 #endif
     }
 
